Templates/helpers.jinja2

```diff
@@ -6,9 +6,9 @@
 } %}
 
 {%- macro print_type(property) -%}
-    {%- if '$ref' in property -%}
+    {%- if property['$ref'] is defined -%}
         std::shared_ptr<{{ property['$ref'].split('/')[-1] }}>
-    {%- elif not 'type' in property -%}
+    {%- elif not property['type'] is defined -%}
```

Comment:
This needs explicit handling for `property` not being a dictionary instead of making sure these attributes are defined. Otherwise, `bool` will be wrapped as `std::shared_ptr<rapidjson::Document>`